plugin/evm/vm_test.go

```diff
@@ -3812,10 +3808,16 @@ func TestSkipChainConfigCheckCompatible(t *testing.T) {
 	genesisWithUpgradeBytes, err := json.Marshal(genesisWithUpgrade)
 	require.NoError(t, err)
 
+	// Reset metrics to allow re-initialization
+	vm.ctx.Metrics = metrics.NewPrefixGatherer()
```

Comment:
nit: should we add a `ResetMetrics` to vm?

Reply:
I've added a `resetMetrics` test only function to do this - better not pollute production code for test only use cases. On top of that, I've migrated comments "// Reset metrics to allow re-initialization" to that new function:

```gp
// resetMetrics resets the vm avalanchego metrics, and allows
// for the VM to be re-initialized in tests.
```